mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/utils/MQTT5ExceptionUtils.java

```diff
@@ -77,4 +81,30 @@ public static void handleSubScribeException(int messageID, Channel channel, Thro
             channel.writeAndFlush(subscribeAckMessage);
         }
     }
+
+    /**
+     * handle publish Exception.
+     *
+     * @param packetId - Mqtt packet exception
+     * @param channel  - Netty Nio channel
+     * @param ex       - exception
+     */
+    public static void handlePublishException(int packetId, Channel channel, Throwable ex) {
+        if (ex instanceof BrokerServiceException.TopicNotFoundException) {
+            MqttMessage unspecifiedErrorPubAckMessage =
+                    MQTTPubAckMessageUtils.createMqtt5(packetId, MqttPubAckReasonCode.TOPIC_NAME_INVALID,
+                            ex.getMessage());
+            channel.writeAndFlush(unspecifiedErrorPubAckMessage);
+        } else if (ex instanceof MQTTNoMatchingSubscriberException) {
+            MqttMessage unspecifiedErrorPubAckMessage =
+                    MQTTPubAckMessageUtils.createMqtt5(packetId, MqttPubAckReasonCode.NO_MATCHING_SUBSCRIBERS,
+                            ex.getMessage());
+            channel.writeAndFlush(unspecifiedErrorPubAckMessage);
+        } else {
+            MqttMessage unspecifiedErrorPubAckMessage =
+                    MQTTPubAckMessageUtils.createMqtt5(packetId, MqttPubAckReasonCode.UNSPECIFIED_ERROR,
+                            ex.getMessage());
+            channel.writeAndFlush(unspecifiedErrorPubAckMessage);
```

Comment:
We can move `channel.writeAndFlush(unspecifiedErrorPubAckMessage)` at the end of this method.

Reply:
sure, i will do that ASAP :)